samtranslator/model/__init__.py

```diff
@@ -345,15 +346,15 @@ def validate_properties_and_return_model(self, cls: Type[RT]) -> RT:
         """
         try:
             return cls.parse_obj(self._generate_resource_dict()["Properties"])
-        except pydantic.error_wrappers.ValidationError as e:
-            error_properties: str = ""
-            with suppress(KeyError):
-                error_properties = ".".join(str(x) for x in e.errors()[0]["loc"])
-            raise InvalidResourceException(self.logical_id, f"Property '{error_properties}' is invalid.") from e
-        except AttributeError as e:
-            raise InvalidResourceException(
-                self.logical_id, "Module 'pydantic.v1' has no attribute 'error_wrappers'"
-            ) from e
+        # TODO there is an issue on pydantic's side where we get an "module 'pydantic.v1' has no attribute 'error_wrappers'"
+        # Once pydantic fixes their issue we should change the bottom line to catch pydantic.error_wrappers.ValidationError instead of all exceptions
+        # except pydantic.error_wrappers.ValidationError as e:
+        #     error_properties: str = ""
+        #     with suppress(KeyError):
+        #         error_properties = ".".join(str(x) for x in e.errors()[0]["loc"])
+        #     raise InvalidResourceException(self.logical_id, f"Property '{error_properties}' is invalid.") from e
+        except Exception as e:
+            raise InvalidResourceException(self.logical_id, "An unexpected error occurred during validation.") from e
```

Comment:
Is there a better error message we can be using here?

Reply:
made it general since its catching all exceptions and its temporary, any suggestions for a better msg?

Reply:
Ohh I see, just read the other comment. In that case it makes sense for a catch all, but I think we could add an extra detail to make it a bit more specific like `An unexpected error occurred when validating resource properties.`